src/websocket/websocket.service.ts

```diff
@@ -0,0 +1,59 @@
+import { createWebSocket } from "./create-websocket";
+import { DatapointRepository } from "./datapoint.repository";
+import { BeforeApplicationShutdown, Injectable } from "@nestjs/common";
+import { WebSocket } from "ws";
+
+@Injectable()
+export class WebsocketService implements BeforeApplicationShutdown {
+  #socket: WebSocket | null;
+  constructor(private dpRep: DatapointRepository) {
+    this.initSocket(
+      "wss://iotnet.teracom.dk/app?token=vnoVQAAAABFpb3RuZXQudGVyYWNvbS5ka0AHfDGv873AtxYtbA-B0Sw",
+    );
+  }
+  beforeApplicationShutdown() {
+    this.#socket?.close();
+  }
+  async initSocket(url: string): Promise<void> {
```

Comment:
Give a homie some breathing room

```suggestion
  }
  
  beforeApplicationShutdown() {
    this.#socket?.close();
  }
  
  async initSocket(url: string): Promise<void> {
```